File: ganapathy_pavers/utils/js/workorder.js
frappe.ui.form.on('Work Order', {
    refresh: function(frm){
        frm.set_query('mould_name', function(frm){
            return {
                filters:{
                    'paver_name': 'production_item'
                }
            }
        })
    }
})

File: ganapathy_pavers/utils/py/workorder.py
from frappe.custom.doctype.custom_field.custom_field import create_custom_fields
def workorder_customization():
        workorder_custom_fields={
        "Work Order":[
            dict(
            fieldname='mould_name',
            label='Mould Name',
            fieldtype='Link',
            options="Asset",
            insert_after='bom_no'
            ),
 
        ]
        }

        create_custom_fields(workorder_custom_fields)
